process.go

```diff
@@ -0,0 +1,179 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"io"
+
+	"github.com/urfave/cli"
+)
+
+// == Direct input processing ========================================
+
+func processInputFiles(c *cli.Context, g *Gomplate) error {
+	inputs, err := readInputs(c.String("in"), c.StringSlice("file"))
+	if err != nil {
+		return err
+	}
+
+	outputs := c.StringSlice("out")
+	if len(outputs) == 0 {
+		outputs = []string{"-"}
+	}
+
+	for n, input := range inputs {
+		if err := renderTemplate(g, input, outputs[n]); err != nil {
```

Comment:
> imo renderTemplate in process.go is maybe not so a bad fit as it is the final step in 'process' (so process could io + render)

Figuring out which files to read and reading them is IMO a separate concern from the render step. Admittedly, `main.go` has a lot of mixed concerns, but if a new file is created I'd rather it be _fairly_ single-responsibility